server/game/gamesteps/revealplots.js

```diff
@@ -61,7 +61,8 @@ class RevealPlots extends BaseStep {
         });
 
         for (let plot of plots) {
-            event.addChildEvent(new Event('onPlotRevealed', { plot: plot }));
+            let plotRevealedEvent = new Event('onPlotRevealed', { plot: plot });
+            event.addChildEvent(plotRevealedEvent);
```

Comment:
I mention in the comment above on recycling to happen within RevealPlots; this is probably the spot for it (and you creating that `plotRevealedEvent` is almost there